### PR TITLE
SUR-274 - Radio group component

### PR DESCRIPTION
**Pull request by BabuTechBsf**
### Description

Added Radio group component
Jira: https://brainstormforce.atlassian.net/browse/SUR-274

### Screenshots

 https://d.pr/v/LRRG99

### Checklist:

-   [ ] My code is tested
-   [ ] My code passes the PHPCS tests
-   [ ] I've created the npm build.
-   [ ] My code follows accessibility standards 
-   [ ] My code has proper inline documentation 
-   [ ] I've included any necessary tests 
-   [ ] I've included developer documentation 
-   [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by jaieds — CHANGES_REQUESTED** [review]
<s>1. The radio button's white circle is not center-aligned. → https://d.pr/i/DNSs6n</s>
2. Some parts of the radio button are not clickable. → https://d.pr/v/zH7EX2
3. No transition for the hover effect.
<s>4. Label overlapping with the radio button → https://d.pr/i/2iNEaG</s>

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -1 +1,206 @@
...
+
+## Step 2: Setup ForceUI
+
+  
+
+1.  **Navigate to the ForceUI directory** (if not already there):
+
+```bash
+
+cd forceui
```
[the remark this answers is not in the document]
Comment (BabuTechBsf) [reply README.md:76]: ok

File: src/utilities/functions.js [hunk 2]
```diff
@@ -11,3 +11,34 @@ export function prefix() {
...
+			return 'grid-cols-9';
+		case 10:
+			return 'grid-cols-10';
+		case 11:
+			return 'grid-cols-11';
+		case 12:
+			return 'grid-cols-12';
+		default:
+			return 'grid-cols-1'; // Fallback to 1 column if an invalid number is provided
+	}
```
[the remark this answers is not in the document]
Comment (BabuTechBsf) [reply src/utilities/functions.js:43]: do we need to change, any pros or cons?

File: src/theme/default-config.js [hunk 3]
```diff
@@ -167,6 +167,16 @@ const defaultTheme = {
...
 				'toggle-off-disabled': '#F3F4F6',
+				// radio-group
+				'radio-button-background': '#FFFFFF',
+				'radio-button-border': '#E5E7EB',
+				'radio-button-hover': '#F3F4F6',
+				'radio-button-selected': '#2563EB',
+				'radio-button-selected-hover': '#1D4ED8',
+				'radio-button-disabled': '#F9FAFB',
+				'radio-button-disabled-border': '#E5E7EB',
+				'radio-button-disabled-background': '#F9FAFB',
```
[the remark this answers is not in the document]
Comment (BabuTechBsf) [reply src/theme/default-config.js:178]: Yes. I discussed the same earlier. I will remove.

File: package.json [hunk 4]
```diff
@@ -47,7 +47,8 @@
     "nanoid": "^5.0.7",
     "styled-components": "^6.1.12",
     "tailwind-merge": "^2.4.0",
-    "tailwindcss-scoped-preflight": "^3.4.3"
+    "tailwindcss-scoped-preflight": "^3.4.3",
+    "wp-scripts": "^0.0.1-security"
```
[the remark this answers is not in the document]
Comment (BabuTechBsf) [reply package.json:51]: Not sure, i didn't install separately, came along with other from project.

[the remark this answers is not in the document]
Comment (vrundakansara) [reply package.json:51]: Let's remove them then

File: src/utilities/functions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vrundakansara) [reply src/utilities/functions.js:43]: Switch is a loop, also we have followed the same pattern everywhere in the library - https://github.com/brainstormforce/force-ui/blob/staging/src/components/badge/badge.jsx#L28

File: src/components/radio-group/readme.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### children
+- **Type:** `string` or `ReactNode`
+- **Description:** The children prop can be used to pass additional content, such as icons or custom elements, that will be displayed within the button.
+
+### Example:
+
+```
+<RadioButton.Group
+    name="food"
```
Comment (ravindra114) [inline src/components/radio-group/readme.md:110]: is this correct? It supposed to be RadioGroup.Group?

**Review by ravindra114 — COMMENTED** [review]
Figma design not matching
https://d.pr/i/1Ax8TX

File: src/utilities/functions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (BabuTechBsf) [reply src/utilities/functions.js:43]: Added

File: src/components/radio-group/readme.md (hunk 5 above)
Comment (BabuTechBsf) [reply src/components/radio-group/readme.md:110]: Fixed

**Review by jaieds — CHANGES_REQUESTED** [review]
@BabuTechBsf, Only these two are not resolved:
1. Some parts of the radio button are not clickable.  → https://d.pr/v/zH7EX2
2. No transition for the hover effect.

**Review by vrundakansara — CHANGES_REQUESTED** [review]
We need to add stories for storybook. @BabuTechBsf

**Review by jaieds — APPROVED** [review]

**Review by jaieds — CHANGES_REQUESTED** [review]
Sometimes I am facing an issue with selection.

Please watch the screencast: https://d.pr/v/R3pR1N

**Comment by BabuTechBsf** [comment]
> Sometimes I am facing an issue with selection.
> 
> Please watch the screencast: https://d.pr/v/R3pR1N

Fixed

**Comment by jaieds** [comment]
@BabuTechBsf, The selection issue seems resolved. ✅

However, in some existing implementations, the UI seems broken. Do we need to handle that case from the plugin side?

**Screenshot:** https://d.pr/i/TcvcX2

**Review by vrundakansara — APPROVED** [review]

**Review by jaieds — APPROVED** [review]

**Pull request merged**